core/tx_pool.go

```diff
@@ -707,8 +719,15 @@ func (pool *TxPool) validateTx(tx types.PoolTransaction, local bool) error {
 		}
 		return ErrInvalidSender
 	}
+
+	inAllowedTxs := false
```

Comment:
By default, if there is no entry for an address in allowedTx list, it should pass. Your current logic makes all txns by default being blacklisted. The correct logic should be 

isNotAllowed := false

if .........
    check whether there is a entry for the sender address
    If Yes:
         check whether the to address matches:
         if Yes:
             isNotAllowed = false
         else
             isNotAllowed = true
    else
         isNotAllowed = false



IsNotAllowed == true is similar to being blacklisted.

Reply:
`inAllowedTxs` indicated whether a tx/address is in the allowed tx whitelist. if it is, then skip to checking blacklist.

Reply:
> `inAllowedTxs` indicated whether a tx/address is in the allowed tx whitelist. if it is, then skip to checking blacklist.

otherwise, transactions will follows the original logic to check if it passes.

Reply:
yeah, we can make it work on its own. but if an address exists in the blacklist and whitelist, only the whitelist takes effect.